scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -371,12 +374,25 @@ public boolean isApproximation() {
       if (x_n_double < dxLow_long || x_n_double > dxHigh_long) {
         throw new DivergenceException("Looking for root out of prescribed domain :[" + dxLow + "," + dxHigh + "]");
       }
-      Duration ff_x_nminus1 = ff.valueAt(x0), ff_x_n = ff.valueAt(x_n);
+      var ff_x_nminus1 = ff.valueAt(x0);
+      if (ff_x_nminus1.noShorterThan(Duration.negate(toleranceYLow)) && ff_x_nminus1.noLongerThan(toleranceYHigh)) {
+        return new RootFindingResult<>(x0, ff_x_nminus1.plus(y), null);
+      }
+      var ff_x_n = ff.valueAt(x_n);
+      if (ff_x_n.noShorterThan(Duration.negate(toleranceYLow)) && ff_x_n.noLongerThan(toleranceYHigh)) {
+        return new RootFindingResult<>(x_n, ff_x_n.plus(y), null);
+      }
+      if (x0.isEqualTo(x1)) {
+        //particular : only one start possible, derivative would be 0
+        throw new NoSolutionException();
+      }
+      history.add(Pair.of(x0, ff_x_nminus1));
+      history.add(Pair.of(x1, ff_x_n));
```

Comment:
We check whether the initial bounds might satisfy the exit criteria. After these checks, we can be sure that if the two bounds are the same, the derivative will be 0, and thus throw an exception.

Reply:
This sounds like it would make a great comment inline with the code. 🤔

Reply:
Agreed. You are so fast, @Twisol :)

Reply:
I have email notifications on and I'm currently a passenger in a vehicle. I've got everything going for me >:)